packages/syft/src/syft/service/code/user_code.py

```diff
@@ -729,6 +739,120 @@ def __call__(self, *args: Any, **kwargs: Any) -> Any:
+
+        for node_id, obj_dict in self.input_policy_init_kwargs.items():
+            # api = APIRegistry.api_for(
+            #     node_uid=node_id.node_id, user_verify_key=node_id.verify_key
+            # )
+            api = APIRegistry.get_by_recent_node_uid(node_uid=node_id.node_id)
+
+            # Creating TwinObject from the ids of the kwargs
+            # Maybe there are some corner cases where this is not enough
+            # And need only ActionObjects
+            # Also, this works only on the assumption that all inputs
+            # are ActionObjects, which might change in the future
+            for _, id in obj_dict.items():
+                mock_obj = api.services.action.get_mock(id)
+                if isinstance(mock_obj, SyftError):
+                    data_obj = api.services.action.get(id)
+                    if isinstance(data_obj, SyftError):
+                        return SyftError(
+                            message="You do not have access to object you want \
+                                to use, or the private object does not have mock \
+                                data. Contact the Node Admin."
+                        )
+                else:
+                    data_obj = mock_obj
+                data_obj.id = id
+                new_obj = ActionObject.from_obj(
+                    data_obj.syft_action_data,
+                    id=id,
+                    syft_node_location=node_id.node_id,
+                    syft_client_verify_key=node_id.verify_key,
+                )
+                res = ep_client.api.services.action.set(new_obj)
+                if isinstance(res, SyftError):
+                    return res
+
+        new_syft_func = deepcopy(self)
+
+        # We are exploring the source code to automatically upload
+        # subjobs in the ephemeral node
+        # Usually, a DS would manually submit the code for subjobs,
+        # but because we dont allow them to interact with the ephemeral node
+        # that would not be possible
+        if "domain" in self.input_kwargs:
+            tree = ast.parse(inspect.getsource(self.local_function))
+            v = LaunchJobVisitor()
+            v.visit(tree)
+            nested_calls = v.nested_calls
+            try:
+                ipython = (
+                    get_ipython()
+                )  # works only in interactive envs (like jupyter notebooks)
+            except Exception:
+                ipython = None
+                pass
+
+            for call in nested_calls:
+                if ipython is not None:
+                    specs = ipython.object_inspect(call)
+                    # Look for nested job locally, maybe we could
+                    # fetch
+                    if specs["type_name"] == "SubmitUserCode":
+                        ep_client.code.submit(ipython.ev(call))
+
+        ep_client.code.request_code_execution(new_syft_func)
```

Comment:
I guess we can just submit and run as admin? In that case we wouldnt need approval